gitops/snapshot.go

```diff
@@ -194,51 +197,57 @@ func PrepareToRegisterIntegrationPipelineRun(snapshot *applicationapiv1alpha1.Sn
 }
 
-// SetSnapshotIntegrationStatusAsFinished sets the HACBS integration status condition for the Snapshot to Finished.
+// SetSnapshotIntegrationStatusAsFinished sets the AppStudio integration status condition for the Snapshot to Finished.
 func SetSnapshotIntegrationStatusAsFinished(snapshot *applicationapiv1alpha1.Snapshot, message string) {
 	condition := metav1.Condition{
-		Type:    HACBSIntegrationStatusCondition,
+		Type:    AppStudioIntegrationStatusCondition,
 		Status:  metav1.ConditionTrue,
-		Reason:  HACBSIntegrationStatusFinished,
+		Reason:  AppStudioIntegrationStatusFinished,
 		Message: message,
 	}
 	meta.SetStatusCondition(&snapshot.Status.Conditions, condition)
 }
 
-// IsSnapshotNotStarted checks if the HACBS Integration Status condition is not in progress status.
+// IsSnapshotNotStarted checks if the AppStudio Integration Status condition is not in progress status.
 func IsSnapshotNotStarted(snapshot *applicationapiv1alpha1.Snapshot) bool {
-	condition := meta.FindStatusCondition(snapshot.Status.Conditions, HACBSIntegrationStatusCondition)
-	if condition == nil || condition.Reason != HACBSIntegrationStatusInProgress {
+	condition := meta.FindStatusCondition(snapshot.Status.Conditions, AppStudioIntegrationStatusCondition)
+	if condition == nil {
+		condition = meta.FindStatusCondition(snapshot.Status.Conditions, LegacyIntegrationStatusCondition)
+	}
+	if condition == nil || condition.Reason != AppStudioIntegrationStatusInProgress {
 		return true
 	}
 	return false
 }
 
-// IsSnapshotValid checks if the HACBS Integration Status condition is not invalid.
+// IsSnapshotValid checks if the AppStudio Integration Status condition is not invalid.
 func IsSnapshotValid(snapshot *applicationapiv1alpha1.Snapshot) bool {
-	condition := meta.FindStatusCondition(snapshot.Status.Conditions, HACBSIntegrationStatusCondition)
-	if condition == nil || condition.Reason != HACBSIntegrationStatusInvalid {
+	condition := meta.FindStatusCondition(snapshot.Status.Conditions, AppStudioIntegrationStatusCondition)
+	if condition == nil {
+		condition = meta.FindStatusCondition(snapshot.Status.Conditions, LegacyIntegrationStatusCondition)
+	}
+	if condition == nil || condition.Reason != AppStudioIntegrationStatusInvalid {
 		return true
 	}
 	return false
 }
 
-// HaveHACBSTestsFinished checks if the HACBS tests have finished by checking if the HACBS Test Succeeded condition is set.
-func HaveHACBSTestsFinished(snapshot *applicationapiv1alpha1.Snapshot) bool {
-	statusCondition := meta.FindStatusCondition(snapshot.Status.Conditions, HACBSTestSuceededCondition)
+// HaveAppStudioTestsFinished checks if the AppStudio tests have finished by checking if the AppStudio Test Succeeded condition is set.
+func HaveAppStudioTestsFinished(snapshot *applicationapiv1alpha1.Snapshot) bool {
+	statusCondition := meta.FindStatusCondition(snapshot.Status.Conditions, AppStudioTestSuceededCondition)
```

Comment:
Do we need to do a similar thing for `HaveAppStudioTestsFinished` and  `HaveAppStudioTestsSucceeded` to check for legacy conditions?

Reply:
Yes, I think we need to have the same thing for those functions.

Reply:
That seems to have done the trick